azure/src/test/java/org/apache/iceberg/azure/adlsv2/ADLSLocationTest.java

```diff
@@ -89,4 +101,17 @@ public void testQuestionMarkInFileName(String path) {
     ADLSLocation location = new ADLSLocation(fullPath);
     assertThat(location.path()).contains(path);
   }
+
+  @ParameterizedTest
+  @CsvSource({
+    "abfs://container@account.dfs.core.windows.net/file.txt, account.dfs.core.windows.net",
+    "abfs://container@account.dfs.core.usgovcloudapi.net/file.txt, account.dfs.core.usgovcloudapi.net",
+    "wasb://container@account.blob.core.windows.net/file.txt, account.blob.core.windows.net",
+    "abfs://account.dfs.core.windows.net/path, account.dfs.core.windows.net",
+    "wasb://account.blob.core.windows.net/path, account.blob.core.windows.net"
+  })
+  void testHost(String path, String expectedHost) {
+    ADLSLocation location = new ADLSLocation(path);
+    assertThat(location.host()).contains(expectedHost);
```

Comment:
Shouldn't this be a strict equals and not just a contains?

Reply:
Good catch!  Fixed in https://github.com/apache/iceberg/pull/11830/commits/8a6498983b0f07ccbaceb9ebe262e0523db7d472